pkg/ingester/instance.go

```diff
@@ -737,8 +737,9 @@ func (i *instance) getStats(ctx context.Context, req *logproto.IndexStatsRequest
 
 	if err = i.forMatchingStreams(ctx, from, matchers, nil, func(s *stream) error {
 		// Consider streams which overlap our time range
+		s.chunkMtx.RLock()
```

Comment:
We read s.chunk.Bounds() in shouldConsiderStreams

Reply:
but `Bounds()` already acquires the RLock [here](https://github.com/grafana/loki/blob/main/pkg/ingester/stream.go#L552). do we still need this?

Reply:
Good spot - we probably don't need this change then. Grabbing the write lock when encoding should be enough since that is what causes the conflict.